forum/users/serializers.py

```diff
@@ -0,0 +1,31 @@
+from rest_framework import serializers
+from rest_framework.validators import UniqueValidator
+
+from users.models import CustomUser
+from users.validators import CustomUserValidator
+
+class UserRegisterSerializer(serializers.ModelSerializer):
+    password = serializers.CharField(write_only=True, required=True)
+    password2 = serializers.CharField(write_only=True, required=True)
+
+    class Meta:
+        model = CustomUser
+        fields = ('user_id', 'first_name', 'last_name', 'email', 'password', 'password2', 'user_phone', 'description',)
```

Comment:
Do you need `password2` here?

Reply:
If we remove the password2 field in the Meta class, it will not be included in the serialized output format and will not be available for validation and there will be errors. And the create method removes password2 from validated_data before creating the user, as this field is not needed to create the user in the model.

Reply:
But we can use validation in `validate()` method
```
    def validate(self, data):
        # Validate that the two password fields match
        if data.get('password') != data.get('password2'):
            raise serializers.ValidationError("Passwords must match.")
        
        CustomUserValidator.validate_password(data.get('password'))
        if data.get('user_phone') is not None:
            CustomUserValidator.validate_user_phone(data.get('user_phone'))

        return data
```

Reply:
I moved the validation of password matching to the `validate()` method, and I still shouldn't remove the `password2` field from the fields because when the Python interpreter gets to the line `if serializer.is_valid()`, we will get this error: 
```
AssertionError: The field 'password2' was declared on serializer UserRegisterSerializer, but has not been included in the 'fields' option.
```